src/main/scala/io/iohk/ethereum/jsonrpc/EthProofService.scala

```diff
@@ -210,7 +210,7 @@ class EthProofService(blockchain: Blockchain, blockGenerator: BlockGenerator, et
     blockParam match {
       case BlockParam.WithNumber(blockNumber) => getBlock(blockNumber).map(ResolvedBlock(_, pendingState = None))
       case BlockParam.Earliest => getBlock(0).map(ResolvedBlock(_, pendingState = None))
-      case BlockParam.Latest => getBlock(blockchain.getBestBlockNumber()).map(ResolvedBlock(_, pendingState = None))
+      case BlockParam.Latest => getBlock(blockchain.getBestSavedBlockNumber()).map(ResolvedBlock(_, pendingState = None))
```

Comment:
Would it perhaps make sense to add a method `getBestBlock` in `Blockchain` to first try to resolve the best block and fallback to best saved block? That removes the duplication of this logic and also the separate steps of "get block number" followed by "fetch block by number."

Reply:
did that already, fixing tests atm